src/Orchard.Web/Modules/Orchard.Layouts/Controllers/BlueprintAdminController.cs

```diff
@@ -43,12 +44,19 @@ public class BlueprintAdminController : Controller, IUpdateModel {
             _shapeFactory = shapeFactory;
             _transactionManager = transactionManager;
             _signals = signals;
+            Services = orchardServices;
+
             T = NullLocalizer.Instance;
         }
 
+        public IOrchardServices Services { get; set; }
```

Comment:
Just a pattern ... sticking with it for the sake of consistency. I'd personally like to have this plugged to autofac like we do for Localizer for instance. Could also be something we add to the `Component` class.